### PR TITLE
DIV-6532-Storing-and-visibility-of-multiple-Servicerefusal-documents

### PR DESCRIPTION
**Pull request by madjava**
# Description

Service application refusal documents can now be multiple files, just like general orders, in this PR the layout has been updated to accommodate this change.

Jira: https://tools.hmcts.net/jira/browse/DIV-6532

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/core/utils/viewHelper.js [hunk 1]
```diff
@@ -21,6 +23,18 @@ const getDownloadableFiles = session => {
...
+/**
+ * The current view content. This is essentially the content.json file but for the specific active language
+ * @param pageInstance The view object
+ * @param session
+ * @returns {{test: string}|{test: string}} The json content object
+ */
+const getCurrentContent = (pageInstance, session) => {
+  const language = session.language || DEFAULT_LANGUAGE;
+  return pageInstance.content.resources[language].translation.content;
+};
```
Comment (madjava) [inline app/core/utils/viewHelper.js:35]: Moving functionality to common location

File: app/steps/service-application-not-approved/content.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
             "deemed": "The court has refused your application for '{{ serviceName }}' that your {{ divorceWho }} received the divorce application.",
             "dispensed": "The court has refused your application to '{{ serviceName }}' that your {{ divorceWho }} doesn’t have to be served the divorce application."
           },
-          "refusalDocumentInfo": "You can read the court’s reasons for this in the <a class=\"govuk-link\" aria-label=\"This link will download the file\" href=\"{{ refusalDocumentUri }}\" target=\"_blank\">{{ refusalDocumentLabel }} (PDF)</a>.",
+          "refusalDocumentInfo": "You can read the court’s reasons for this in the '{{ refusalDocumentLabel }}' document in 'Download your documents'. </a>.",
```
Comment (madjava) [inline app/steps/service-application-not-approved/content.json:13]: Updating view content based on AC

File: app/steps/service-application-not-approved/content.json [hunk 3]
```diff
@@ -123,7 +123,7 @@
             "deemed": "[CY] The court has refused your application for '{{ serviceName }}' that your {{ divorceWho }} received the divorce application.",
             "dispensed": "[CY] The court has refused your application to '{{ serviceName }}' that your {{ divorceWho }} doesn’t have to be served the divorce application."
           },
-          "refusalDocumentInfo": "[CY] You can read the court’s reasons for this in the <a class=\"govuk-link\" aria-label=\"This link will download the file\" href=\"{{ refusalDocumentUri }}\" target=\"_blank\">{{ refusalDocumentLabel }} (PDF)</a>.",
+          "refusalDocumentInfo": "[CY] You can read the court’s reasons for this in the '{{ refusalDocumentLabel }}' document in 'Download your documents'</a>.",
```
Comment (madjava) [inline app/steps/service-application-not-approved/content.json:126]: Ditto

File: app/steps/service-application-not-approved/index.js [hunk 4]
```diff
@@ -49,38 +49,22 @@ module.exports = class ServiceApplicationNotApproved extends Step {
...
-      .filter(file => {
-        return file.type === serviceApplicationFile;
-      })
-      .map(file => {
-        return {
-          fileUri: file.uri,
-          fileLabel: this.getRefusalDocumentLabel(session, serviceApplicationFile)
-        };
-      })[0];
-  }
```
Comment (madjava) [inline app/steps/service-application-not-approved/index.js:75]: Logic no longer needed

File: app/steps/service-application-not-approved/index.test.js [hunk 5]
```diff
@@ -74,16 +85,18 @@ describe(modulePath, () => {
...
+      multipleOccurringTypes.forEach(item => {
         // eslint-disable-next-line max-nested-callbacks
-        testCustom(done, agent, underTest, [], response => {
-          const numberOfItems = 2;
-          expect(getGeneralOrderOccurrencesInPage(response)).to.equal(numberOfItems);
+        it(`should have two '${item.viewLabel} (PDF)' labels in template view`, done => {
+          // eslint-disable-next-line max-nested-callbacks
+          testCustom(done, agent, underTest, [], response => {
+            expect(getOccurrencesInPage(response, `${item.viewLabel} (PDF)`)).to.equal(item.expectedCount);
+          });
```
Comment (madjava) [inline app/steps/service-application-not-approved/index.test.js:99]: Reusable feature so this block of logic is not repeated over an over

File: package.json [hunk 6]
```diff
@@ -30,7 +30,7 @@
     "test-e2e": "NODE_ENV=testing LOG_LEVEL=OFF NODE_PATH=. ./node_modules/.bin/codeceptjs run-multiple parallel -c test/end-to-end/codecept.conf.js --steps --reporter mocha-multi",
     "test-overnight-e2e": "NODE_ENV=testing LOG_LEVEL=OFF NODE_PATH=. codeceptjs run -c test/end-to-end/codecept.conf.js --steps",
     "test-crossbrowser-e2e": "NODE_ENV=testing LOG_LEVEL=OFF NODE_PATH=. node_modules/.bin/codeceptjs run-multiple  ${BROWSER_GROUP:-'--all'} -c test/end-to-end/saucelabs.conf.js --steps --reporter mocha-multi",
-    "lint": "NODE_PATH=. eslint ."
+    "lint": "NODE_PATH=. eslint . --fix"
```
Comment (madjava) [inline package.json:33]: Updated the config here so build does not have to fail due to trivial issues e.g. space etc, if lint can fix it then let it fix and move on. Time saver, idea from CCD 🤓

File: app/steps/service-application-not-approved/index.js [hunk 7]
```diff
@@ -49,38 +49,22 @@ module.exports = class ServiceApplicationNotApproved extends Step {
   setDocumentInfo(session, ctx) {
     session.downloadableFiles = this.getDownloadableFiles(session);
     ctx.serviceName = this.getServiceName(session);
-    const { fileLabel, fileUri } = this.getServiceRefusalDocument(session);
-    ctx.refusalDocumentLabel = fileLabel;
-    ctx.refusalDocumentUri = fileUri;
+    ctx.refusalDocumentLabel = this.getRefusalDocumentLabel(session);
```
Comment (madjava) [inline app/steps/service-application-not-approved/index.js:52]: Generating file label based on service type

**Review by Dizrupto — APPROVED** [review]

**Review by yarpo — APPROVED** [review]

**Review by yarpo — APPROVED** [review]

**Review by Dizrupto — APPROVED** [review]

**Pull request merged**